docs/Roadmap.md

```diff
@@ -0,0 +1,140 @@
+# The Lean Standard Library Roadmap
+
+## Vision
+
+Lean needs a mature standard library that is designed for the entire
+Lean user base.  Particular communities, such as the math community, may
+have their own common libraries, but standard will provide the common
+set of definitions, theorems, and tactics to facilitate collaboration
+across the Lean user base.
+
+Standard will be the default library for Lean, and be a key part of the
+user's experience.  Standard definitions and documentation must be
+designed with new users in mind as tutorials and other new user
+documentation should be able to assume Std is in scope.  Std must be an
+exemplar library for writing good documentation, testing and other
+extensions in Lean.  Std must stay up to date with Lean including
+support for all Lean release candidates. Std must also provide a stable
+interface so packages can efficiently upgrade to new versions.
+
+The scope of the standard library includes the following areas:
+
+ * Tactics for **proof automation**
+ * **Data structure libraries** for functional and
+   concurrent programming.
+ * Monadic transformers and **verification support**.
+ * **Broadly useful typeclasses** including with and without lemmas.
+ * **IO operations** for concurrency, file system operations, launching
+   new processes and networking.  The IO layer should support both
+   synchronous and asynchronous operations.
+ * **Usability extension mechanisms** including commands for finding lemmas,
+   providing hints, linting, and and other more speculative automation.
+ * **Metaprogramming operations** that help build new tactics.
+
+## Type Classes Standardization.
+
+Lean needs typeclasses for common properties such as associativity so
+that tactics that depend on common properties can be writen
+independently of the operations that satisfy those properties.  Mathlib
+provides many of the classes, but the classes may be more specialized
+than a standard library for non-mathematicians may want.  The standard
```

Comment:
I recommend avoiding the term "non-mathematicians". The following rewording focuses on the actions rather than personal characteristics.
```suggestion
provides many of the classes, but the classes tend to be specialized for mathematical purposes rather than programming and other development purposes.  The standard
```